omod-2.4/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs2_4/TaskDefinitionResource2_4.java

```diff
@@ -0,0 +1,164 @@
+	public DelegatingResourceDescription getRepresentationDescription(Representation rep) {
+		if (rep instanceof DefaultRepresentation) {
+			DelegatingResourceDescription description = new DelegatingResourceDescription();
+			description.addProperty("uuid");
+			description.addProperty("name");
+			description.addProperty("description");
+			description.addProperty("taskClass");
+			description.addProperty("startTime");
+			description.addProperty("started");
+			description.addLink("ref", ".?v=" + RestConstants.REPRESENTATION_REF);
+			description.addLink("full", ".?v=" + RestConstants.REPRESENTATION_FULL);
+			description.addSelfLink();
+			return description;
+		} else if (rep instanceof FullRepresentation) {
+			DelegatingResourceDescription description = new DelegatingResourceDescription();
+			description.addProperty("uuid");
+			description.addProperty("name");
+			description.addProperty("description");
+			description.addProperty("taskClass");
+			description.addProperty("startTime");
+			description.addProperty("lastExecutionTime");
+			description.addProperty("repeatInterval");
+			description.addProperty("startOnStartup");
+			description.addProperty("startTimePattern");
+			description.addProperty("started");
+			description.addProperty("properties");
+			description.addSelfLink();
+			description.addLink("ref", ".?v=" + RestConstants.REPRESENTATION_REF);
+			return description;
+		} else if (rep instanceof RefRepresentation) {
+			DelegatingResourceDescription description = new DelegatingResourceDescription();
+			description.addProperty("uuid");
+			description.addProperty("name");
+			description.addProperty("description");
+			description.addProperty("taskClass");
+			description.addSelfLink();
+			return description;
+		}
+		return null;
+	}
+
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingResource#getCreatableProperties()
+	 */
+	@Override
+	public DelegatingResourceDescription getCreatableProperties() {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		description.addRequiredProperty("name");
+		description.addRequiredProperty("description");
+		description.addRequiredProperty("taskClass");
+		description.addRequiredProperty("startTime");
+		description.addRequiredProperty("repeatInterval");
+		description.addRequiredProperty("startOnStartup");
+		description.addRequiredProperty("properties");
+		return description;
+	}
+
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.DelegatingCrudResource#save(java.lang.Object)
+	 */
+	@Override
+	public TaskDefinition save(TaskDefinition taskDefinition) {
+		taskServiceWrapper.saveTaskDefinition(taskDefinition);
+		return taskDefinition;
+	}
+
+	@Override
+	public TaskDefinition getByUniqueId(String uniqueId) {
+		TaskDefinition taskDefinition = taskServiceWrapper.getTaskByName(uniqueId);
+		if (taskDefinition == null) {
+			taskDefinition = taskServiceWrapper.getTaskByUuid(uniqueId);
```

Comment:
I can see why we want to *add* the functionality to get a task by UUID, but do we also want to remove the functionality to get the task by ID?

Reply:
@ibacher yes i think its ideal to get a task by its UUID and leave the task by ID for internal database identification and operations